racket/src/racket/src/vector.c

```diff
@@ -1076,6 +1107,28 @@ static Scheme_Object *unsafe_vector_star_set (int argc, Scheme_Object *argv[])
   return scheme_void;
 }
 
+static Scheme_Object *unsafe_vector_star_cas (int argc, Scheme_Object *argv[])
+{
+  Scheme_Object *vec = argv[0];
+  Scheme_Object *idx = argv[1];
+  Scheme_Object *ov = argv[2];
+  Scheme_Object *nv = argv[3];
+
+#ifdef MZ_USE_FUTURES
+  return mzrt_cas((volatile size_t *)(SCHEME_VEC_ELS(vec) + SCHEME_INT_VAL(idx)),
+                  (size_t)ov, (size_t)nv)
```

Comment:
Instead of `size_t`, I'd prefer to use `uintptr_t`, since the rest of the implementation uses that type for pointer-sized integers.

Reply:
I agree. I changed it in the box-cas! implementation too.